utils/s3/client.cc

```diff
@@ -1107,7 +1107,7 @@ class client::do_upload_file : private multipart_upload {
             req._headers["x-amz-tagging"] = seastar::format("{}={}", _tag->key, _tag->value);
         }
         req.write_body("bin", len, [f = std::move(f)] (output_stream<char>&& out_) mutable {
-            auto input = make_file_input_stream(std::move(f), input_stream_options());
+            auto input = make_file_input_stream(f, input_stream_options());
```

Comment:
how is this related to adding a proxy?

Reply:
adding a proxy and start using it with injected errors and retrying failed request set this problem afloat

Reply:
i just don't really understand why we cannot `std::move()` it ? because it does not compile anymore because

> adding a proxy and start using it with injected errors and retrying failed request set this problem afloat

still cannot connect the dots..

Reply:
but what exactly is the "problem"? i am not against or for this change. just wanted to understand why.

Reply:
coz retry replays the request which calls `write_body` which calls `make_file_input_stream` which moved the `file` into it, meaning second call will use already moved argument

Reply:
> do you think it is better to move this change to the "Add retries" commit?

yes.